students/pinstripezebra/mailroomP3

```diff
@@ -0,0 +1,128 @@
+"""
+Created on Sat Feb 10 08:23:07 2018
+
+@author: seelc
+"""
+import datetime
+
+#Dictionary containing all donor data
+donationHistory = {'William Gates, III':[10312.3, 3],
+                   'Mark Zuckerberg':[31245.24, 3],
+                   'Jeff Bezos':[9432.01, 2],
+                   'Paul Alan':[7244.35, 2],
+                   'Elon Musk':[13148.94, 1],
+        }
+
+def main():
+    #promting the user to choose a main action
+    userAction = ""
+    while userAction !="4":
+        userAction = input( "Choose an action: \n1- Send a thank you\n2-Create a Report\n3-Send a thank you to everyone\n4-Quit\n")
+        #Attempting to throw an exception if userAction isnt 1, 2, or 3
+        try:
+            assert userAction =="1" or "2" or "3" or "4"
+            #if the user chooses to send a thank you
+            if userAction == "1":
+                nameRequest = input("Please select a name: ")
+                #0 stands in for donationAmount, purpose of having second input
+                #is to make the thankYou function easier to test
+                thankYou(nameRequest, 0)
+        
+            #if the user chooses to create a report
+            elif userAction == "2":
+                printReport(donationHistory)
+        
+            #if the user chooses to send a thank you to everyone
+            elif userAction == "3":
+                groupThanks(donationHistory)
+            #if none of the above options are chosen quits the program
+            elif userAction == 4:
+                print("Quiting program")
+                break
+        except:
+            print("Encountered an error")
+    
+            
+    
+if __name__ == '__main__':
```

Comment:
This goes at the very bottom of your program so that the python interpreter can run through and identify all your functions first.